agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/BeforeAgentInstaller.java

```diff
@@ -187,6 +196,53 @@ public void run() {
                 }
             }
         });
+
+        pollJsonConfigEveryMinute();
+    }
+
+    private static volatile Long lastModifiedTime;
+    private static void pollJsonConfigEveryMinute() {
+        Executors.newSingleThreadScheduledExecutor(ThreadPoolUtils.createDaemonThreadFactory(JsonConfigPolling.class))
+                .scheduleWithFixedDelay(new JsonConfigPolling(), 60, 60, SECONDS);
+    }
+
+    private static class JsonConfigPolling implements Runnable {
```

Comment:
i think this would be good class to extract out into top-level, to avoid cluttering `BeforeAgentInstaller`

you can track `lastModifiedTime` as instance field

and would be great if we can get `lastModifiedTime` from before the initial loading (add another field to `Configuration`?), that way we avoid a potential race condition of the config file being updated between when we read it the first time, and when this thread starts checking for updates